mod/state-transition/pkg/core/metrics.go

```
@@ -0,0 +1,45 @@
+	return &stateProcessorMetrics{
+		sink: sink,
+	}
+}
+
+func (s *stateProcessorMetrics) gaugeTimestamps(
+	payloadTimestamp uint64,
+	consensusTimestamp uint64,
+) {
+	// the diff can be positive or negative depending on whether the payload
+	// timestamp is ahead or behind the consensus timestamp
+	diff := int64(payloadTimestamp) - int64(consensusTimestamp) //#nosec:G701
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int64

(gosec)